src/OpenLoco/src/Map/MapGenerator/PngTerrainGenerator.cpp

```diff
@@ -12,40 +12,38 @@ using namespace OpenLoco::Diagnostics;
 
 namespace OpenLoco::World::MapGenerator
 {
-    void PngTerrainGenerator::generate(const S5::Options& options, const fs::path& path, HeightMapRange heightMap)
+    void PngTerrainGenerator::generate(const S5::Options& options, const fs::path& path, HeightMap& heightMap)
     {
         if (!fs::is_regular_file(path))
         {
-            Logging::error("Can't find terrain file ({})", path);
+            Logging::error("Can't find heightmap file ({})", path);
             return;
         }
 
         auto pngImage = PngOps::loadPng(path.string());
         if (pngImage == nullptr)
         {
-            Logging::error("Can't load terrain file ({})", path);
+            Logging::error("Can't load heightmap file ({})", path);
             return;
         }
 
         const int maxHeightmapLevels = 64 - options.minLandHeight;
         const float scalingFactor = maxHeightmapLevels / 255.f;
 
-        for (int32_t y = 0; y < World::kMapRows; y++)
+        std::fill_n(heightMap.data(), heightMap.size(), options.minLandHeight);
+
+        auto width = std::min((int)World::kMapColumns, pngImage->width);
+        auto height = std::min((int)World::kMapRows, pngImage->height);
+
+        for (int32_t y = 0; y < height; y++)
         {
-            for (int32_t x = 0; x < World::kMapColumns; x++)
+            for (int32_t x = 0; x < width; x++)
             {
-                if (y >= pngImage->height || x >= pngImage->width)
-                {
-                    heightMap[{ x, y }] = options.minLandHeight;
-                }
-                else
-                {
-                    png_byte red, green, blue, alpha;
-                    pngImage->getPixel(x, y, red, green, blue, alpha);
-
-                    auto imgHeight = std::max({ red, green, blue });
-                    heightMap[{ x, y }] = options.minLandHeight + (imgHeight * scalingFactor);
-                }
+                png_byte red, green, blue, alpha;
+                pngImage->getPixel(x, y, red, green, blue, alpha);
+
+                auto imgHeight = std::max({ red, green, blue });
+                heightMap[{ y, x }] += imgHeight * scalingFactor; // this must be { y, x } otherwise the heightmap is mirrored
```

Comment:
I'm still not convinced this is correct. Swapping x,y to y,x seems to me like it would mirror on the diagonal axis instead. Could you perhaps illustrate what's going wrong? (e.g. by using a non-symmetric letter, such as P)

Elsewhere, I see the heightmap being iterated over using 512-x instead, i.e. back to front. This suggests it's just the horizontal axis that needs mirroring.

Reply:
heightmap:
![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/070c7fcc-7d63-4681-bfec-82da30112547)

`{ x, y }`
![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/f40dde34-9fb2-49bd-a0d4-9e2beb8c4c09)

`{ y, x }`
![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/c54ac1ca-736c-4038-be2a-b2e33483c266)

I guess the question I need to ask is, should the top of the heightmap (y = 0) be the 'left' side  or the 'right' side of the openloco map

Reply:
The grass tile is { x=1, y=1 } and the rock tile is { x=8, y=4 }
![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/aeba8033-8e12-43f2-b794-81b9aad9864b)